src/compute/merge_sort/mod.rs

```diff
@@ -291,6 +291,29 @@ where
             None => self.right = None,
         }
     }
+
+    /// Collect the MergeSortSlices to be a vec for reusing
+    #[warn(dead_code)]
+    pub fn to_vec(self, limit: Option<usize>) -> Vec<MergeSlice> {
+        match limit {
+            Some(limit) => {
+                let mut v = Vec::with_capacity(limit);
+                let mut current_len = 0;
+                for (index, start, len) in self {
+                    v.push((index, start, len));
+
+                    if len + current_len >= limit {
```

Comment:
I think that this misses the last entries: if the last slice is past `limit`, imo it should be modified to be equal to `limit - current_len`, so that we do not lose the last items within the slice